src/bindings.rs

```diff
@@ -91,4 +94,14 @@ impl Preprocessor {
             Err(err) => Err(as_javascript_error(err, self.core.source_map()).into()),
         }
     }
+
+    pub fn parse(&self, src: String, filename: Option<String>) -> Result<JsValue, JsValue> {
+        let result = self
+            .core
+            .parse(&src)
+            .map_err(|_err| self.process(src, filename).unwrap_err())?;
```

Comment:
This calls `process` just to get the error, eventually it can be extracted out into some common method, but this works for now